tests/utils/logging_tester.py

```diff
@@ -0,0 +1,79 @@
+from functools import wraps
+import os
+import json
+
+
+def _check_logs(reference_logfile, generated_logfile):
+    reference_logs = open(reference_logfile, "r").readlines()
+    generated_logs = open(generated_logfile, "r").readlines()
+
+    if len(reference_logs) != len(generated_logs):
+        msg_assert = ""
+        if len(reference_logs) > len(generated_logs):
+            diff = len(reference_logs) - len(generated_logs)
+            msg_assert += f"Logs generated are SHORTER than the expected logs. There are {diff} extra logs.\n"
+            msg_assert += "Last log of the generated log is : \n"
+            msg_assert += generated_logs[-1]
+        else:
+            diff = len(generated_logs) - len(reference_logs)
+            msg_assert += f"Logs generated are LONGER than the expected logs.\n There are {diff} extra logs :\n"
+            for log in generated_logs[len(reference_logs) :]:
+                msg_assert += log
+        assert 0, msg_assert
+
+    for index, (ref, gen) in enumerate(zip(reference_logs, generated_logs)):
+        # As they are string, we only need to check if they are equal. If they are not, we then compute a more precise difference, to debug.
+        if ref == gen:
+            continue
+        ref_log = json.loads(ref)
+        gen_log = json.loads(gen)
+        diff_keys = [
+            d1[0] for d1, d2 in zip(ref_log.items(), gen_log.items()) if d1[1] != d2[1]
+        ]
+        # \n does not work in f-strings.
+        newline = "\n"
+        assert (
+            len(diff_keys) == 0
+        ), f"Lgos don't match at {index} st log. : \n{newline.join([f'In {key} field, got -> {gen_log[key]}. {newline}Expected : -> {ref_log[key]}.' for key in diff_keys])}"
+
+
+def logs_comparison(control_data_file, log_path_from_media_dir):
+    """Decorator used for any test that needs to check logs.
+
+    Parameters
+    ----------
+    control_data_file : :class:`str`
+        Name of the control data file, i.e. .log that will be compared to the outputed logs.
+        .. warning:: You don't have to pass the path here.
+
+    log_path_from_media_dir : :class:`str`
+        The path of the .log generated, from the media dir. Example: /logs/Square.log.
+    """
+
+    def decorator(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            # NOTE : Every args goes seemingly in kwargs instead of args; this is perhaps Pytest.
+            result = f(*args, **kwargs)
+            tmp_path = kwargs["tmp_path"]
+            tests_directory = os.path.dirname(
+                os.path.dirname(os.path.abspath(__file__))
+            )
+            path_control_data = os.path.join(
+                tests_directory, "control_data", "logs_data", control_data_file
+            )
```

Comment:
Nitpick: one variable is `path_control_data` while the other one is `control_data_file`. Why not `control_data_path` and `control_data_file`?

Reply:
I don't underswtand your comment. 

One is the file name and the other is the entire path.

Reply:
he means the names should follow a better pattern

Reply:
Ow yea, forgive my stupidity. I wil change this rn